Replace plt.show() with fig.show() in the docs

jhaigh0 opened the pull request:
### Description of work

Since upgrading to matplotlib 3.7, plt.show() has sometimes created freezing / hanging behaviour when run from within workbench. This pr replaces `plt.show()` uses in the documentation with `fig.show()` to make sure all scripts in the docs are compatible with workbench. 
We'd still like to find the route cause of this problem, but there is no harm in this change to the docs which will hopefully stop as many people encountering it.

This was suggested in the comments of https://github.com/mantidproject/mantid/issues/36905 

### To test:

 - cmake with docs plot option on. I could only make the  docs build work on linux with this option
   - `cmake -DDOCS_PLOTDIRECTIVE=ON --preset=linux`
 -  build `docs-html`
 - Check the files which have an uncomented `fig.show()` within a `.. plot::`
   - Sometimes the plots weren't showing up on the first build for me (irrespective of the show method), only after I made an edit to the code. Unsure what to suggest in this case ofther than putting a comment or something after the `fig.show()` to prompt it to re build
 - Check the scripts I've changed a bit more significantly run in mantid

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jhaigh0 committed:
Switch to using fig.show() from plt.show()

Since upgrading to matplotlib 3.7, plt.show() has sometimes created freezing / hanging behaviour when run from within workbench

jhaigh0 commented on docs/source/tutorials/python_in_mantid/solutions/02_pim_sol.rst:
```diff
@@ -88,14 +64,14 @@ B - Plotting ILL Data
     axes.legend() #.draggable()
 
-    #plt.show()
+    #fig.show()
```
Not too sure why, if you run in mantid it still adds a plot to the plots manager, so I guess It's less annoying to have them hidden by default.

GuiMacielPereira replied:
Ah I see, makes sense

jhaigh0 commented on docs/source/fitting/fitfunctions/BivariateGaussian.rst:
```diff
@@ -86,16 +86,12 @@ Here is an example of fitting a 2D histogram:
-    plt.title('Fit')
-    plt.show()
-
+    fig, axes = plt.subplots(nrows=1, ncols=2, subplot_kw={'projection': 'mantid'})
+    axes[0].imshow(Z, origin='lower')
+    axes[0].set_title('Data')
+    axes[1].imshow(ZFit, origin='lower')
+    axes[1].set_title('Fit')
+    fig.show()
```
Do you mean the two plots side by side were the same? (so `Z` and `ZFit` are the same data?)

GuiMacielPereira replied:
Yes, but it looks like this was already like that?

jhaigh0 replied:
would that make sense since ZFit is trying to be as close as possible to Z?

GuiMacielPereira replied:
Yes, I just made a commit to increase the noise in the original data so that it's clearer that one is the fit of the other

GuiMacielPereira committed:
Increased noise in data for clearer comparison with fit

- Previous noise was so small that Z and ZFit looked identical
- With increased noise it's clear that one is a smooth fit to the noisy data
